apps/web/src/utils/redirectIfNameDoesNotExist.ts

```diff
@@ -3,19 +3,45 @@ import {
   getBasenameAddress,
   getBasenameEditor,
   getBasenameOwner,
+  isBasenameInGracePeriod,
 } from 'apps/web/src/utils/usernames';
 import { redirect } from 'next/navigation';
+import { logger } from 'apps/web/src/utils/logger';
 
 export async function redirectIfNameDoesNotExist(username: Basename) {
-  const [address, editor, owner] = await Promise.all([
-    getBasenameAddress(username),
-    getBasenameEditor(username),
-    getBasenameOwner(username),
-  ]).catch(() => {
-    redirect(`/name/not-found?name=${username}`);
-  });
-  // Domain does not have address or editor (ie: doesn't exist)
-  if (!address || !editor || !owner) {
-    redirect(`/name/not-found?name=${username}`);
+  let address, editor, owner;
+  let apiError = false;
+
+  try {
+    [address, editor, owner] = await Promise.all([
+      getBasenameAddress(username),
+      getBasenameEditor(username),
+      getBasenameOwner(username),
+    ]);
+  } catch (error) {
+    logger.error('Error fetching basename address, editor, or owner', {
+      error,
+      username,
+    });
+    apiError = true;
+  }
+
+  // If API calls failed OR returned null/undefined values, the name doesn't exist or is expired
+  const nameNotFound = apiError || !address || !editor || !owner;
```

Comment:
I think it's redundant to check `apiError` here since if there was an error, then at least one of the three values are guaranteed to be undefined
```suggestion
  const nameNotFound = !address || !editor || !owner;
```